core/src/test/scala/integration/kafka/api/PlaintextConsumerSubscriptionTest.scala

```diff
@@ -227,11 +228,29 @@ class PlaintextConsumerSubscriptionTest extends AbstractConsumerTest {
 
   @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumAndGroupProtocolNames)
   @MethodSource(Array("getTestQuorumAndGroupProtocolParametersAll"))
-  def testSubscribeInvalidTopic(quorum: String, groupProtocol: String): Unit = {
-    // Invalid topic name due to space
-    val invalidTopicName = "topic abc"
+  def testSubscribeInvalidTopicCanUnsubscribe(quorum: String, groupProtocol: String): Unit = {
+    val consumer = createConsumer()
+
+    setupSubscribeInvalidTopic(consumer)
```

Comment:
Create a followup Jira https://issues.apache.org/jira/browse/KAFKA-17647.

Reply:
Hey here, just for the record, this test covers the scenario because even though `setupSubscribeInvalidTopic` will loop until the `InvalidTopicException` is consumed (once), the invalid topic remains in the consumer subscription (until unsubscribe), so it continues to be included in the Metadata requests, therefore the network poll continues to find the `InvalidTopicException` in the metadata response and propagates it. This test fails without this PR, and passes with this it. 

On the close the story is slightly different, the test was only flaky without this PR (because the test initially wasn't calling close, it was only the test infra that would call close when tearing down the whole suite).

Makes sense? If so, I would say we don't need KAFKA-17647 right?

Reply:
@lianetm Thanks for sharing your thoughts. My point was that the `ErrorEvent` (`InvalidTopicException`) might not exist when calling `unsubscribe`, as the invalidTopics could have already been cleared by a previous poll call [0]. In such a case, `testSubscribeInvalidTopicCanUnsubscribe` could pass without needing this PR. Additionally, a simple fix for this test case would be to add a small delay after `setupSubscribeInvalidTopic` to ensure that the `invalidTopics` is created by the network poll.

[0] https://github.com/apache/kafka/blob/05696037d330e5e355d62cb1b977bac9df78db82/clients/src/main/java/org/apache/kafka/clients/Metadata.java#L638